test/atomics-unshared.wast.from-wast

```diff
@@ -1,7 +1,7 @@
 (module
  (type $none_=>_none (func))
  (memory $0 1 1)
- (func $foo
+ (func $foo (type $none_=>_none)
```

Comment:
Can we not print these if GC is not on? That is, even if the type system is isorecursive, we don't need these types if the features don't require it. It would also make the diff here much smaller I think.

Reply:
Yeah, I added that logic and it helped some of the lit tests, but the problem is that most of these tests run with `-all`.